tests/sapt-exch-disp-inf/hcn.input

```diff
@@ -0,0 +1,34 @@
+memory 4 gb
```

Comment:
Can you combine these tests into a single test for each molecule? If you examine other SAPT tests they should give you a good template to work off of. There is also the `CMakeLists.txt` that should be copied over and modified from these test examples as well and in `psi4/tests/CMakeLists.txt:40` please add the name of this test. We usually try to keep alphabetical order there.

Reply:
I combined them, but I had to resolve an issue to do it. The reference was getting changed from 'RHF' to 'RKS' during the first run, so the initial calculation would work but the subsequent one would fail at psi4/driver/procrouting/sapt/sapt_proc.py:106. I altered the if statement there to look at the global reference, but I'm not sure if that is the preferred resolution to that.

Reply:
Good work finding a solution. But actually that was an existing problem. Please revert that change to sapt_proc and add a line `optstash.restore()` to https://github.com/psi4/psi4/pull/1490/files#diff-e167218b440b023fab5287792ffc4ef3L258 and see if that fixes the problem.

Reply:
That works. Thanks